src/genomic_features/annotate_anndata.py

```diff
@@ -0,0 +1,112 @@
+import warnings
+
+import pandas as pd
+
+
+def annotate_anndata(
+    adata_var: pd.DataFrame,
+    annotation_df: pd.DataFrame,
+    on: str = None,
```

Comment:
I found the variable names a bit confusing because in pandas syntax `on` means that the column is present in both tables. Maybe we could name it `left_on` and `right_on` like in `pd.merge`